app.ts

```diff
@@ -0,0 +1,309 @@
+		try {
+			const response = await this.fetchData('http://localhost:2050/columns');
+			const res = JSON.parse(response);
+			this.columnNames = res.map((columnName: any) => ({ name: columnName }));
+			this.data = new Array<GridData>(this.columnNames.length);
+		} catch (error) {
+			console.error('Failed to fetch columns', error);
+			throw error;
+		}
+	}
+	//get records from API for fetch an search functionality 
+	async fetchAndProcessRecords(from: number = this.firstVal, to: number = this.lastVal) {
+		try {
+			$('#spinner').show()
+			$('#grid').hide()
+			const response = await this.fetchData(`http://localhost:2050/records?from=${from}&to=${to}`);
+			const res = JSON.parse(response);
+			const processedData = res.map((record: any) => {
+				const obj: GridData = {};
+				for (let j = 0; j < this.columnNames.length && j < record.length; j++) {
+					const columnName = this.columnNames[j].name;
+					const columnValue = record[j];
+					obj[columnName] = columnValue;
+				}
+				return obj;
+			});
+			$('#spinner').hide();
+			$('#grid').show()
+			return processedData;
+		} catch (error) {
+			console.error('Failed to fetch records', error);
+			throw error;
+		}
+	}
+
+
+	async fetchRecords() {
+		const maxRange = this.totalItems - 1;
+		const from = this.firstVal;
+		let to = Math.min(from + this.pageSize, maxRange);
+		if (to >= maxRange) {
+			this.currentPage = Math.floor(maxRange / this.pageSize) + 1; // Set currentPage to the last page
+			to = maxRange;
+		}
+		try {
+			const processedData = await this.fetchAndProcessRecords(from, to);
+			this.data = processedData;
+			this.displayRecords();
+			this.updatePageInfo();
+		} catch (error) {
+			alert('failed to fetch records ');
+		}
+	}
+
+
+	async searchRecords(searchValue: number) {
+		try {
+			const maxRange = this.totalItems - 1; // Maximum allowed Value
+			if (searchValue >= 0 && searchValue <= maxRange) {
+				const from = searchValue;
+				const to = Math.min(from + this.pageSize, maxRange);
+				const processedData = await this.fetchAndProcessRecords(from, to);
+				this.data = processedData;
+				this.currentPage = Math.ceil(from / this.pageSize) + 1
+				this.firstVal = from; // Set firstVal to searched value
+				this.lastVal = from + this.pageSize; // Calculate lastVal based on pageSize
+				this.displayRecords();
+				this.updatePageInfo();
+			}
+			else {
+				alert('Please enter values in the range (0-999999)');
+			}
+		} catch (error) {
+			alert('Failed to fetch records');
+		}
+	}
+	adjustGridHeight() {
+		const gridElement = document.getElementById('grid');
+		const pageCntrl = $('.grid-controls').innerHeight();
+		const screenHeight = $(window).innerHeight();
+		if (gridElement && pageCntrl !== undefined && screenHeight !== undefined) {
+			this.maxGridHeight = screenHeight - pageCntrl;
+			gridElement.style.height = `${this.maxGridHeight}px`;
+			gridElement.style.overflow = 'none';
+		}
+	}
+
+	private async fetchData(url: string): Promise<any> {
```

Comment:
we dont like making use of any too much, maybe give the types 
for example:
```suggestion
	private async fetchData(url: string): Promise<number> | Promise<string> {
```